lua/treesitter-context/render.lua

```diff
@@ -6,49 +6,71 @@ local config = require('treesitter-context.config')
 
 local ns = api.nvim_create_namespace('nvim-treesitter-context')
 
--- Don't access directly, use get_bufs()
-local gutter_bufnr --- @type integer?
-local context_bufnr --- @type integer?
-
-local gutter_winid --- @type integer?
-local context_winid --- @type integer?
-
---- @param buf integer?
---- @return integer buf
-local function create_buf(buf)
-  if buf and api.nvim_buf_is_valid(buf) then
-    return buf
-  end
-
-  buf = api.nvim_create_buf(false, true)
+--- @class WindowContext
+--- @field bufnr integer The buffer number
+--- @field gutter_bufnr integer The gutter buffer number
+--- @field context_bufnr integer The context buffer number
+--- @field gutter_winid integer? The window ID of the gutter
+--- @field context_winid integer? The window ID of the context
+local WindowContext = {}
+WindowContext.__index = WindowContext
 
-  vim.bo[buf].undolevels = -1
-  vim.bo[buf].bufhidden = 'wipe'
+local windowContexts = {} --- @type table<integer, WindowContext>
 
-  return buf
+--- @param winid integer?
+local function win_close(winid)
+  vim.schedule(function()
```

Comment:
The code doesn't show why this is needed.

Reply:
I move win_close to top since store_context needs to call close if underline buffer changed